### PR TITLE
[ampproject/amphtml] ♿ [Story a11y] aria-label on amp-story-360 canvas

### PR DESCRIPTION
**Pull request by processprocess**
Contributes to #32493
- Adds `role="image"` to canvas element.
- Propagates `aria-label` from child element to canvas element.
- Adds `alt-text` to example for demo purposes.

### CODE REVIEW

File: extensions/amp-story-360/0.1/amp-story-360.js [hunk 1]
```diff
@@ -352,7 +357,18 @@ export class AmpStory360 extends AMP.BaseElement {
...
+    const altText = dev()
+      .assertElement(
+        this.element.querySelector('amp-img') ||
+          this.element.querySelector('amp-video')
+      )
+      .getAttribute('alt-text');
+    this.canvas_ = createElementWithAttributes(
+      this.element.ownerDocument,
+      'canvas',
+      dict({'role': 'img', 'aria-label': altText})
```
Comment (mszylkowski) [inline extensions/amp-story-360/0.1/amp-story-360.js:369]: Is there a `role="video"` or similar?

Comment (processprocess) [reply extensions/amp-story-360/0.1/amp-story-360.js:371]: > Is there a role="video" or similar?

There isn't. I think `img` is the closest item from the [list of roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques).

> Also, I wonder if alt-text is already read by screen readers or not, becuase we don't want to read that twice.

Do you mean read once from the original media element, then again from the canvas? 
We hide the original media element. The screen reader does not have access to it which is why we need to put it on the canvas.
I added a comment in the code to help explain that.

**Review by mszylkowski — APPROVED** [review]
👍

**Comment by gmajoulet** [comment]
Can you explain how publishers should use this? `alt-text` is a custom attribute and thus invalid AMP, publishers can't specify it as far as I understand. Plus it's not documented anywhere.
Why don't we use the existing HTML accessibility features and propagate, in this order of preference: `alt`, `title`, `aria-label`?

**Pull request merged**